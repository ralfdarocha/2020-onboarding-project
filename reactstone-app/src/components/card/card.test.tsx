import React from 'react';
import { shallow } from 'enzyme';
import { Card } from '@components/index';
import cardsMock from '@mock/cards.json'
import renderer from 'react-test-renderer';

describe('<Card />', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <Card card={{
                "cardId": "HERO_07",
                "dbfId": "893",
                "name": "Gul'dan",
                "cardSet": "Basic",
                "type": "Hero",
                "faction": "Neutral",
                "rarity": "Free",
                "health": 30,
                "collectible": true,
                "playerClass": "Warlock",
                "img": "http://media.services.zam.com/v1/media/byName/hs/cards/enus/HERO_07.png",
                "locale": "enUS"
            }} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders image fallback', () => {
        const tree = renderer.create(
            <Card card={{
                cardId: "HERO_08f",
                cardSet: "Hero Skins",
                collectible: true,
                dbfId: "61598",
                faction: "Neutral",
                health: 30,
                locale: "enUS",
                name: "Scholar Jaina",
                playerClass: "Mage",
                rarity: "Free",
                type: "Hero"
            }} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('shouldn\'t have an image', () => {
        const { img, ...card } = cardsMock[0];
        const component = shallow(<Card card={card} />);
        expect(component.find('img').exists()).toBeFalsy();
    });

});
