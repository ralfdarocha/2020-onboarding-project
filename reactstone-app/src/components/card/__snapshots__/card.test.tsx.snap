// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<div
  className="card card-hero"
>
  <span
    className="card-image"
    style={
      Object {
        "display": "inline-block",
        "height": "387",
        "width": "256",
      }
    }
  />
</div>
`;

exports[`<Card /> renders image fallback 1`] = `
<div
  className="card card-hero no-image"
>
  <div
    className="no-image-card"
  >
    <div
      className="card-image"
    >
      Image not found
    </div>
    <div
      className="card-cost"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="card-title"
    >
      <span>
        Scholar Jaina
      </span>
    </div>
    <div
      className="card-health"
    >
      <span>
        30
      </span>
    </div>
  </div>
</div>
`;
